File: my-submissions/0048-rotate-image/solution.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& arr) {
        int n = arr.size();
        //transpose of matrix
        for(int i = 0;i<n;i++){
            for(int j = 0;j<=i;j++){
                swap(arr[i][j],arr[j][i]);
            }
        }
        //swap the columns
        for(int i = 0;i<n;i++){
             reverse(arr[i].begin(),arr[i].end());
        }
    }
};

File: my-submissions/0048-rotate-image/solution.java
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        for(int i = 0;i<n;i++){
            for(int j =i+1;j<n;j++){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        int row = 0;
        int l = 0;
        int r = n -1;
        while(row<n){
            while(l<r){
                int temp = matrix[row][l];
                matrix[row][l] = matrix[row][r];
                matrix[row][r] = temp;
                l++;
                r--;
            }
            l = 0;
            r = n-1;
            row++;
        }
        return;
    }
}
